src/main/java/de/dhbw/bahn/schicht_0_plugins/http/EinfacherHttpServer.java

```diff
@@ -81,40 +82,59 @@ public void handle(HttpExchange exchange) {
-                return EventTyp.LESEN;
-            case "PUT":
-                return EventTyp.AKTUALISIEREN;
-            case "POST":
-                return EventTyp.ERSTELLEN;
-            case "DELETE":
-                return EventTyp.LOESCHEN;
-            default:
-                return null;
-        }
-    }
-
     private EventAntwort verarbeiteAnfrage(HttpExchange exchange) throws IOException {
         System.out.println(exchange.getRequestMethod() + " Anfrage an " + exchange.getHttpContext().getPath());
 
+        if (!wirdAnfragemethodeUnterstuetzt(exchange))
+            return baueEventAntwortAnfragemethodeNichtUnterstuetzt();
+
+        Event event = baueEventAus(exchange);
+        if (!istEventRueckrufVorhandenFuer(event))
+            return baueEventAntwortEventRueckrufNichtGefunden();
+
+        EventAntwort antwort = rufeEventRueckrufAufFuer(event, exchange);
+        return antwort;
+    }
+
+    private boolean wirdAnfragemethodeUnterstuetzt(HttpExchange exchange) {
+        Optional<EventTyp> eventTyp = holeEventTypFuer(exchange);
+        return eventTyp.isPresent();
+    }
+
+    private Optional<EventTyp> holeEventTypFuer(HttpExchange exchange) {
+        String httpMethod = exchange.getRequestMethod();
+        return HttpMethodeZuEventTypMapping.fuer(httpMethod);
+    }
+
+    private EventAntwort baueEventAntwortAnfragemethodeNichtUnterstuetzt() {
+        return new EventAntwort(404, "Diese Anfragemethode wird nicht unterstützt", MimeTyp.SCHLICHT);
+    }
+
+    private Event baueEventAus(HttpExchange exchange) {
+        EventTyp eventTyp = holeEventTypFuer(exchange).get();
         String pfad = exchange.getHttpContext().getPath();
+        Event event = new Event(pfad, eventTyp);
+        return event;
+    }
 
-        EventTyp eventTyp = httpMethodeZuEventTyp(exchange.getRequestMethod());
-        if (eventTyp == null)
-            return new EventAntwort(404, "Diese Anfragemethode wird nicht unterstützt", MimeTyp.SCHLICHT);
-        Event route = new Event(pfad, eventTyp);
-        if (!this.rueckrufTabelle.containsKey(route)) {
-            return new EventAntwort(404, "Not Found", MimeTyp.SCHLICHT);
-        }
-        EventRueckruf rueckruf = this.rueckrufTabelle.get(route);
+    private boolean istEventRueckrufVorhandenFuer(Event event) {
+        return this.rueckrufTabelle.containsKey(event);
+    }
+
+    private EventAntwort baueEventAntwortEventRueckrufNichtGefunden() {
+        return new EventAntwort(404, "Not Found", MimeTyp.SCHLICHT);
+    }
 
+    private EventAntwort rufeEventRueckrufAufFuer(Event event, HttpExchange exchange) throws IOException {
+        EventRueckruf rueckruf = holeEventRueckrufFuer(event);
         String query = exchange.getRequestURI().getQuery();
         Map<String, String> parameter = leseParameter(query);
         String koerper = leseKoerper(exchange.getRequestBody());
+        return rueckruf.bearbeiteAnfrage(event, koerper, parameter);
```

Comment:
Hier könntest du den Wert noch in einer Variable zwischenspeichern. Bzw könntest du auch oben sonst Werte direkt zurückgeben.